app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java

```diff
@@ -189,22 +277,38 @@ private Element getGuiElement() {
      *
      * We need to iterate through XML nodes manually, as mConfig.getDocumentElement() will also
      * return nested elements inside folder element.
+     *
+     * Returns if changes to the config have been made.
      */
-    private void changeLocalDeviceName() {
+    private boolean changeLocalDeviceName() {
         NodeList childNodes = mConfig.getDocumentElement().getChildNodes();
         for (int i = 0; i < childNodes.getLength(); i++) {
             Node node = childNodes.item(i);
             if (node.getNodeName().equals("device")) {
+                Log.i(TAG, "changeLocalDeviceName: Current device name updated to build.MODEL.");
                 ((Element) node).setAttribute("name", Build.MODEL);
+
+                /**
+                  * Only alter the first occurency of the device tag in assumption it is the device running this instance.
+                  * Reason: Sometimes encountered a bug when manually restored syncthing db+config with root explorer
+                  *         where app started and all devices got their name altered to the model name. Config was messed up then.
+                  *         This should work around the problem at least until a better fix will be implemented.
+                  * ToDo:   Implement a better fix by reading the device ID first, e.g. from REST API and only alter the correct
+                  *         node's device name attribute.
+                  */
+                return true;
```

Comment:
Maybe put this in a seperate PR? And I would put the long explanation into the javadoc for the method, not inside the function. Or just describe the problem in a github issue and add the issue link.